responses/responses_test.go

```diff
@@ -15,3 +15,117 @@ limitations under the License.
 */
 
 package responses_test
+
+import (
+	"encoding/json"
+	"github.com/RedHatInsights/insights-operator-utils/responses"
+	//"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+)
+
+var mock_payload = map[string]interface{}{
+	"numbers_l": []int{1, 2, 3, 5, 8},
+	"color_s":   "blue",
+	"extra_data_m": map[string]interface{}{
+		"param1": 1,
+		"param2": false,
+		"param3": 0.75,
+	},
+}
+
+// Test BuildResponse that returns simple map with key "status" and given value
+func TestBuildResponse(t *testing.T) {
+	status_str := "I'm a teapot"
+	expected_response := map[string]interface{}{
+		"status": status_str,
+	}
+	response := responses.BuildResponse(status_str)
+	if equal := reflect.DeepEqual(expected_response, response); !equal {
+		t.Errorf("Expected response %v", expected_response)
+	}
+}
+
+// Test BuildResponse that returns simple map with key "status" and value "ok"
+func TestBuildOkResponse(t *testing.T) {
+	expected_response := map[string]interface{}{
+		"status": "ok",
+	}
+	response := responses.BuildOkResponse()
+	if equal := reflect.DeepEqual(expected_response, response); !equal {
+		t.Errorf("Expected response %v", expected_response)
+	}
+}
+
+// Test BuildResponse that returns simple map with key status: ok and given data
+func TestBuildOkResponseWithData(t *testing.T) {
+	expected_response := map[string]interface{}{
+		"data":   mock_payload,
+		"status": "ok",
+	}
+	response := responses.BuildOkResponseWithData("data", mock_payload)
+	if equal := reflect.DeepEqual(expected_response, response); !equal {
+		t.Errorf("Expected response %v", expected_response)
+	}
+}
+
+func TestSendResponse(t *testing.T) {
+	test_server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		response := responses.BuildOkResponseWithData("data", mock_payload)
+		responses.SendResponse(w, response)
+	}))
+	defer test_server.Close()
+
+	res, err := http.Get(test_server.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var response map[string]interface{}
+
+	json.NewDecoder(res.Body).Decode(&response)
+	expected_response := map[string]interface{}{
+		"data":   mock_payload,
+		"status": "ok",
+	}
+
+	t.Log(response)
+	t.Log(expected_response)
+	// why aren't they equal???
+	if equal := reflect.DeepEqual(expected_response, response); !equal {
+		t.Errorf("Expected response %v.", expected_response)
+	}
```

Comment:
Guys @tisnik @bond95 I don't understand this. The printed response and expected response look the same, but the DeepEqual check says they arent equal. Am I missing something?

Reply:
I'd need to debug it, lemme check

Reply:
I solved it a little differently as not to waste too much time on just unit tests.
We can discuss tomorrow